java/plasma/src/main/java/org/apache/arrow/plasma/PlasmaClient.java

```diff
@@ -35,13 +36,27 @@
 
   private final long conn;
 
+  private PlasmaClientException plasmaClientException = null;
+
   protected void finalize() {
-    PlasmaClientJNI.disconnect(this.conn);
+    if (plasmaClientException == null) {
+      PlasmaClientJNI.disconnect(this.conn);
```

Comment:
It's ok to solve the issue we met, but this may bring memory leak issue in cpp layer. We need figure out some way in later PR.